poi/src/main/java/org/apache/poi/ss/formula/FormulaParser.java

```diff
@@ -1182,8 +1182,6 @@ private SheetIdentifier parseSheetName(boolean isSndPartOfQuotedRange) {
                         if (look == '\''){
                             // Any single quotes which were already present in the sheet name will be converted to double single quotes ('')
                             // so switch back to single quote
-                            sb.appendCodePoint(look);
-                            nextChar();
```

Comment:
It's not necessary to do it here because on this step variable "look" has second single quote of the pair of single quotes and will be appended to string builder on next iteration of the loop